src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -480,54 +476,89 @@ static X509Certificate2 PersistKey(X509Certificate2 fullCertificate)
                 var certificateBytes = fullCertificate.Export(X509ContentType.Pkcs12, "");
                 return new X509Certificate2(certificateBytes, "", X509KeyStorageFlags.DefaultKeySet);
             }
+
+            static X509Certificate2 LoadCertificateKey(X509Certificate2 certificate, string keyPath, string password)
+            {
+                var keyText = File.ReadAllText(keyPath);
+                return TryAttachPemRSAKey(certificate, keyText, password) ??
+                    TryAttachPemDSAKey(certificate, keyText, password) ??
+                    TryAttachPemECDSAKey(certificate, keyText, password);
+            }
+
+            static X509Certificate2 GetCertificate(CertificateConfig certInfo, string certificatePath, string certificateKeyPath)
+            {
+                if (X509Certificate2.GetCertContentType(certificatePath) != X509ContentType.Unknown)
```

Comment:
The PEM format certs should come back as X509, not Unknown, meaning this should always go into this if and never hit the later return.

Reply:
I changed it to check explicitly for the Cer type and load the key separately in both cases.